core/arch/x86/decode.c

```diff
@@ -2173,6 +2162,202 @@ decode_predicate_from_instr_info(uint opcode, const instr_info_t *info)
     return DR_PRED_NONE;
 }
 
+opnd_size_t
+decode_get_input_size_from_opcode(int opcode, bool evexw)
+{
+    /* See Intel Vol.2A 2.6.5 Compressed Displacement (disp8*N) Support in EVEX. When it's
+     * not possible determine the input size based on evex.W, we use the opcode.
+     */
+    switch (opcode) {
+    case OP_vpextrb:
+    case OP_vpinsrb:
+    case OP_vpbroadcastb: return OPSZ_1;
+    case OP_vpinsrw:
+    case OP_vpextrw:
+    case OP_vpbroadcastw: return OPSZ_2;
+    case OP_vcvtss2si:
+    case OP_vcvttss2si:
+    case OP_vcomiss:
+    case OP_vcvtss2usi:
+    case OP_vcvttss2usi: return OPSZ_4;
+    case OP_vcvtsd2si:
+    case OP_vcvttsd2si:
+    case OP_vcomisd:
+    case OP_vcvtsd2usi:
+    case OP_vcvttsd2usi: return OPSZ_8;
```

Comment:
How about a XXX comment about encoding this into the tuple type to avoid this dispatch on what is pure static info which could come from the decode table instead of needing yet another lookup here.

Reply:
Sure, but a XXX suggests that this needs to be fixed, and I wouldn't suggest to fix it even if I had time. Because I think this would make it harder to debug and to extend should Intel add anything to the instruction set. I'll add it.

Reply:
One could add another flag to the decode_table but not sure if it's worth it? Adding comment anyway...

Reply:
Then include your opinion in the comment.  From an external point of view, having a single piece of info instead of re-deriving what is static and never changes looks like something to improve, so a comment explaining the downsides seems worthwhile.

Reply:
I thought about thatiis more. The "input size" is not clearly specified by Intel, e.g. in 2.6.5. Vol. 2A,. in particular, there is no clear table entry per instruction which input size they are. So the fact that we're doing that based on opcode is made up by me and it just happens to be like that. That said, it actually may be better to specify based on encoding, i.e. in the decode_table. Folding it into the tuple type seems bad, but I think if I am adding just 4 new flags DR_INPUT_OPSZ_1, DR_INPUT_OPSZ_2, DR_INPUT_OPSZ_4, DR_INPUT_OPSZ_8, that wouldn't hurt and get rid of the function. WDYT. Doing this now.

Reply:
OK, sure.  Maybe qualify the name a little more: DR_EVEX_INPUT_OPSZ_1?  DR_INPUT_ELEMENT_OPSZ_1?

Reply:
Done.